gatling-jms/src/main/scala/io/gatling/jms/action/JmsRequestSend.scala

```diff
@@ -0,0 +1,133 @@
+    val actor = system.actorOf(JmsRequestSendActor.props(attributes, protocol, statsEngine, next))
+    new ExitableActorDelegatingAction(genName("jmsRequest"), statsEngine, next, actor)
+  }
+}
+
+object JmsRequestSendActor {
+
+  def props(attributes: JmsAttributes, protocol: JmsProtocol, statsEngine: StatsEngine, next: Action) =
+    Props(new JmsRequestSendActor(attributes, protocol, statsEngine, next))
+}
+
+class JmsRequestSendActor(attributes: JmsAttributes, protocol: JmsProtocol, val statsEngine: StatsEngine, val next: Action)
+    extends ValidatedActionActor {
+
+  // Create a client to refer to
+  val client = JmsClient(protocol, attributes.destination)
+
+  /**
+   * Framework calls the execute() method to send a single request
+   * <p>
+   * Note this does not catch any exceptions (even JMSException) as generally these indicate a
+   * configuration failure that is unlikely to be addressed by retrying with another message
+   */
+  def executeOrFail(session: Session): Validation[Unit] = {
+
+    val messageProperties = resolveProperties(attributes.messageProperties, session)
+
+    val jmsType = resolveOptionalExpression(attributes.jmsType, session)
+
+    // sendtime
+    val startDate = nowMillis
+
+    val msg = jmsType.flatMap(jmsType =>
+      messageProperties.flatMap { props =>
+        attributes.message match {
+          case BytesJmsMessage(bytes) => bytes(session).map(bytes => client.sendBytesMessage(bytes, props, jmsType))
+          case MapJmsMessage(map) => map(session).map(map => client.sendMapMessage(map, props, jmsType))
+          case ObjectJmsMessage(o) => o(session).map(o => client.sendObjectMessage(o, props, jmsType))
+          case TextJmsMessage(txt) => txt(session).map(txt => client.sendTextMessage(txt, props, jmsType))
+        }
+      })
+
+    // done time
+    val endDate = nowMillis
+
+    msg.map { msg =>
+      executeNext(session, startDate, endDate, OK, next, attributes.requestName)
+    }
+
+  }
+
+  private def executeNext(
+    session: Session,
+    sent: Long,
+    received: Long,
+    status: Status,
+    next: Action,
+    title: String,
+    message: Option[String] = None) = {
+
+    val timings = ResponseTimings(sent, received)
+    statsEngine.logResponse(session, title, timings, status, None, message)
+    next ! session.logGroupRequest(timings.responseTime, status).increaseDrift(nowMillis - received)
+
+  }
+
+  def resolveProperties(
+    properties: Map[Expression[String], Expression[Any]],
+    session: Session): Validation[Map[String, Any]] = {
+    properties.foldLeft(Map.empty[String, Any].success) {
+      case (resolvedProperties, (key, value)) =>
+        val newProperty: Validation[(String, Any)] =
+          for {
+            key <- key(session)
+            value <- value(session)
+          } yield key -> value
+
+        for {
+          newProperty <- newProperty
+          resolvedProperties <- resolvedProperties
+        } yield resolvedProperties + newProperty
```

Comment:
This methode is from JmsRepReply...so maybe because key and value are expressions? But not sure why is this like it is...
I am also not sure what to change to keep the function....could you help me with that?